shared/fs/header/add-new.js

```diff
@@ -0,0 +1,141 @@
+// @flow
+import * as React from 'react'
+import {isMobile, globalStyles, globalColors, globalMargins, platformStyles} from '../../styles'
+import {Box, ClickableBox, Icon, Text, type IconType} from '../../common-adapters'
+import FloatingMenu, {
+  FloatingMenuParentHOC,
+  type FloatingMenuParentProps,
+} from '../../common-adapters/floating-menu'
+
+type AddNewProps = {
+  style?: Object,
+  showText: boolean,
+  menuItems: Array<{
+    onClick: () => void,
+    icon: IconType,
+    title: string,
+  }>,
+  pathElementsNoKeybase: Array<string>,
+}
+
+const getIcon = (tlfType: string): IconType => {
+  switch (tlfType) {
+    case 'private':
+      return 'icon-folder-private-16'
+    case 'public':
+      return 'icon-folder-public-16'
+    case 'team':
+      return 'icon-folder-team-16'
+    default:
+      return 'iconfont-question-mark'
+  }
+}
+
+const header = (pathElementsNoKeybase: Array<string>) => {
```

Comment:
I'd prefer not to do this `pathElementsNoKeybase` thing. In all cases you have to slice it anyway (or take the first element), so you might as well just pass around the whole set of `pathElements`.